munit/src/main/scala-0.21/munit/internal/Compat.scala

```diff
@@ -0,0 +1,25 @@
+package munit.internal
+
+import munit.Location
+import scala.quoted._
+import scala.reflect.Selectable.reflectiveSelectable
+
+object Compat {
+  def productElementNames(p: Product): Iterator[String] =
+    p.productElementNames
+  def collectionClassName(i: Iterable[_]): String = {
+    i.asInstanceOf[{ def collectionClassName: String }].collectionClassName
+  }
+
+  trait LocationMacro {
+    inline implicit def generateLocation: Location = ${ locationImpl() }
+  }
+
+  def locationImpl()(given qctx: QuoteContext): Expr[Location] = {
+    import qctx.tasty.{_, given}
+    val path = rootPosition.sourceFile.jpath.toString
+    val startLine = rootPosition.startLine
+    '{ new Location(${Expr(path)}, ${Expr(startLine)}) }
+  }
```

Comment:
So this is quite straightforward, but it seems that the expansion site is slightly different than Scala 2 macros, so some tests fail because the `startLine` is off. I've tried playing around with by-1 fixes, but it's not that simple.